pkg/api/v1beta3/types.go

```diff
@@ -781,7 +781,7 @@ type PodSpec struct {
 	// Optional: Set DNS policy.  Defaults to "ClusterFirst"
 	DNSPolicy DNSPolicy `json:"dnsPolicy,omitempty" description:"DNS policy for containers within the pod; one of 'ClusterFirst' or 'Default'"`
 	// NodeSelector is a selector which must be true for the pod to fit on a node
-	NodeSelector map[string]string `json:"nodeSelector,omitempty" description:"selector which must match a node's labels for the pod to be scheduled on that node"`
+	NodeSelector string `json:"nodeSelector,omitempty" description:"selector which must match a node's labels for the pod to be scheduled on that node"`
```

Comment:
Unless there is some trickery going on that I am missing - this is not a compatible change.

docs/devel/api_changes.md

I don't think you can make this change this way.  You can add a new field with a new name and write the logic to handle both modes, but you can't just change it.